packages/js/github-actions/actions/phpcs-diff/action.yml

```diff
@@ -0,0 +1,32 @@
+name: PHPCS Diff
+description: Run PHPCS to the changed lines of code, set error annotations to a pull request.
+
+runs:
+  using: composite
+  steps:
+    # Checkout repository.
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+
+    # Set up PHP and tools.
+    - uses: woocommerce/grow/prepare-php@actions-v1-pre
+
+    #Log information
+    - shell: bash
+      run: |
+        vendor/bin/phpcs -i
+        vendor/bin/diffFilter -v
+
+    # Run PHPCS with diffFilter in the coverageChecker.
+    - shell: bash
+      run: |
+        vendor/bin/diffFilter --phpcsStrict <(git diff HEAD^...HEAD) <(vendor/bin/phpcs ./* -q --report=json) --report=phpcs 0 > /tmp/phpcs-diff.json
```

Comment:
:grey_question: Do I get it right, that we check here only the changes made in the last commit? not everything that was made in this PR?

Do those annotations accumulate somehow? co we would see the reports from the changes made in first commits?

Reply:
When creating a PR on GitHub, it will create a merge commit locally based on the `head` and `base` branches of the PR. The merge commit is referenced like `refs/pull/<number>/merge`. It's the same in GitHub Actions, when triggering a workflow to a pull request, it uses the revision of a merge commit to run the workflow as well. Therefore, `git diff HEAD^...HEAD` includes all diffs of a PR.

I intended to write a code comment on it, but I forgot. Added in https://github.com/woocommerce/grow/pull/17/commits/9f20f5e628ef1bc35a8196c7db77a31169eb4e55.

Reply:
To confirm this, I pushed two commits that each had a code change in different files to the RP 458-gh-woocommerce-points-and-rewards.

![image](https://user-images.githubusercontent.com/17420811/174943894-172ef3ca-4b79-4fa9-a9f9-7d6286bc1f9f.png)

The annotations were shown in both files.

![image](https://user-images.githubusercontent.com/17420811/174943995-92f3ff6c-4675-4943-97c2-fddbd47af604.png)

Reply:
Awesome, thanks for explanation.